Automattic/wp-calypso
# Plans 2023: Migrate plan-type-selector under plans-grid package

Related to https://github.com/Automattic/wp-calypso/issues/81117

## Proposed Changes

- Updates dependencies and migrates the `plan-type-selector` under `plans-grid` for enabling reuse in comparison grid.
- No other functional or design changes. The `plan-type-selector` works as before and is exported from the `plans-grid` package. There is room for cleanup, and some of it will happen in follow-up PR that will add the toggle to comparison grid's header.
- ~This PR contains changes from https://github.com/Automattic/wp-calypso/pull/84098 and https://github.com/Automattic/wp-calypso/pull/84050, which should be reviewed and merged first, then rebase here.~ - Done

## Testing Instructions



* Go to `/plans/[ free site ]` and `/start/plans` and ensure toggle works as before.
* Go to `/start/onboarding-pm` and ensure toggle shows 1-year/2-year toggle
* Changing interval works and updates the URL with respective term e.g. `/plans/monthly|yearly|etc`
* Switching to `monthly` in `/start/plans` will show a tootlip with the savings info next to the toggle

## Pre-merge Checklist
